devlab/tests/base.py

```diff
@@ -274,14 +277,30 @@ def check_snapshot_state(self, snp_id):
         return False
 
     @staticmethod
+    def check_floating_ip_assigned(vm, floating_ip_address):
+        for net in vm.addresses:
+            for addr in vm.addresses.get(net):
+                if addr.get('OS-EXT-IPS:type') == 'floating':
+                    return True
+        try:
+            vm.add_floating_ip(floating_ip_address)
+        except nova_exceptions.BadRequest as e:
+            LOG.warning("Floating IP is already assigned to VM: %s", e)
+        return False
+
+    @staticmethod
     def wait_until_objects_created(obj_list, check_func, timeout):
         obj_list = obj_list[:]
         waiting = 0
         delay = 1
         while waiting < timeout:
             for obj in obj_list[:]:
-                if check_func(obj):
-                    obj_list.remove(obj)
+                if isinstance(obj, tuple):
```

Comment:
In this case we can't just remove tuple from obj_list, if originally it was not a list of tuples